### PR TITLE
Feature/contextless

### PR DESCRIPTION
**Pull request by st3v3nhunt**
There is a bit of work to do around removing the duplication but the bulk of the changes should be good for review.

### CODE REVIEW

**Comment by st3v3nhunt** [comment]
Ready for full review

File: app/middleware/setLocals.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+function fromRequest(req, res, next) {
+  const location = req.query.location;
```
Comment (neilbmclaughlin) [inline app/middleware/setLocals.js]: consider direct assign to res.locals.context/res.locals.location?

File: test/integration/app.js [hunk 2]
```diff
@@ -1,29 +1,25 @@
...
-  expect(res).to.be.html;
-}
-
 describe('redirection', () => {
   it('default should be find help page', (done) => {
     chai.request(server)
       .get('/')
       .end((err, res) => {
-        checkHtmlResponse(err, res);
+        iExpect.checkHtmlResponse(err, res);
```
Comment (neilbmclaughlin) [inline test/integration/app.js]: 😄  Nice use of 'i'

Would read well as iExpect.htmlResponse200() or iExpect.htmlResponse({ expectedStatus: 200 });

File: test/integration/app.js [hunk 3]
```diff
@@ -34,194 +30,358 @@ describe('redirection', () => {
...
+        expect(err).to.not.be.equal(null);
+        expect(res).to.have.status(404);
+        // eslint-disable-next-line no-unused-expressions
+        expect(res).to.be.html;
 
         const $ = cheerio.load(res.text);
 
-        expect($('.local-header--title').text())
-          .to.equal('Stomach ache');
+        expect($('.local-header--title--question').text().trim())
```
Comment (neilbmclaughlin) [inline test/integration/app.js]: Just for info I've raised an issue regarding the class naming https://github.com/ermlikeyeah/cat-static-dev/issues/2

File: app/middleware/prerender.js [hunk 4]
```diff
@@ -2,19 +2,29 @@ const mapLink = require('../lib/mapLink');
...
   const open = req.query.open || 'true';
+  const context = res.locals.context;
   const location = res.locals.location;
-  let altResultsUrl = '';
+  let openParam = '';
+  let headerMessage = '';
 
   if (open === 'true') {
-    altResultsUrl = `${res.locals.SITE_ROOT}/results?location=${location}&open=false`;
+    openParam = 'false';
```
Comment (st3v3nhunt) [inline app/middleware/prerender.js]: Sure, better readability.

File: app/middleware/setLocals.js (hunk 1 above)
Comment (st3v3nhunt) [inline app/middleware/setLocals.js]: I was going to do that but didn't as I thought it was more clear this way. Obviously not.

File: test/integration/app.js (hunk 2 above)
Comment (st3v3nhunt) [inline test/integration/app.js]: Yeah, that is better. I'll sort that.

**Pull request merged**